symengine/polys.h

```diff
@@ -60,9 +69,9 @@ RCP<const Polynomial> sub_poly(const Polynomial &a, const Polynomial &b);
 //! Multiplying two Polynomial a and b
 RCP<const Polynomial> mul_poly(RCP<const Polynomial> a, RCP<const Polynomial> b);
 
-inline RCP<const Polynomial> polynomial(const vec_symbol &vars, hash_set &&polys_set)
+inline RCP<const Polynomial> polynomial(const vec_symbol &vars, hash_set &polys_set)
 {
-    return make_rcp<const Polynomial>(vars, std::move(polys_set));
+    return make_rcp<const Polynomial>(vars, polys_set);
```

Comment:
You have not used move constructor in your tests. Make the change there too, if you want to use it